### PR TITLE
Supporting C-interop with array indices

### PR DESCRIPTION
**Pull request by czgdp1807**
https://github.com/lcompilers/lpython/issues/609

### CODE REVIEW

**Comment by czgdp1807** [comment]
@certik This is ready as well.

**Review by certik — APPROVED** [review]
I think this is fine.

File: integration_tests/bindc_03.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+@ccall
+def g(a: CPtr, value: i32) -> None:
+    pass
+
+@ccall
+def get_array(size: i32) -> CPtr:
+    pass
+
+@ccallable
```
Comment (certik) [inline integration_tests/bindc_03.py:11]: Since this is not called from C, it doesn't have to be `@ccallable` I think.

Comment (czgdp1807) [reply integration_tests/bindc_03.py:11]: Yes. I think I just took the example as it is from #609.

Comment (czgdp1807) [reply integration_tests/bindc_03.py:11]: Since its merged do you want `ccallable` to be removed in the next PR.

**Comment by certik** [comment]
Sure, if things work. Those are small changes, right now it's important for me to get these features in.

**Pull request merged**